Makefile

```
@@ -12,51 +13,59 @@ all: test manager
 # Run tests
 test: generate fmt vet unit
 
+.PHONY: unit
 unit: manifests unit-test
 
+.PHONY: unit-test
 unit-test:
 	go test ./pkg/... ./cmd/... -coverprofile cover.out
 
 # Run against the configured Kubernetes cluster in ~/.kube/config
+.PHONY: run
 run: generate fmt vet
 	go run ./cmd/manager/main.go
 
 # Install CRDs into a cluster
+.PHONY: install
 install: manifests
 	kubectl apply -f vendor/github.com/openshift/machine-api-operator/install
 	kubectl apply -f config/crds
-
-# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
-deploy: manifests
-	cat provider-components.yaml | kubectl apply -f -
+	kustomize build config | kubectl apply -f -
 
 # Generate manifests e.g. CRD, RBAC etc.
+.PHONY: manifests
 manifests:
 	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd paths=./pkg/apis/... output:crd:dir=./config/crds/
-	kustomize build config/ > provider-components.yaml
+	@sed -i '/^    controller-gen.kubebuilder.io\/version: (devel)/d' config/crds/*
 
 # Run go fmt against code
+.PHONY: fmt
 fmt:
 	go fmt ./pkg/... ./cmd/...
 
 # Run go vet against code
+.PHONY: vet
 vet:
 	go vet ./pkg/... ./cmd/...
 
 # Generate code
+.PHONY: generate
 generate:
-ifndef GOPATH
-	$(error GOPATH not defined, please define GOPATH. Run "go help gopath" to learn more about GOPATH)
-endif
 	go generate ./pkg/... ./cmd/...
 
+.PHONY: generate-check
+generate-check: generate manifests
```

I think we don't want this target to depend on `manifests` because that will mean we run it twice, and the check script will never see any changes. It should probably not have any dependencies at all, to be safe.
```suggestion
generate-check:
```

---

Made the change in the latest commit. Thanks Doug.